File: hrms-frontend/src/services/onboardingService.js
// src/services/onboardingService.js
import api from '../api/axiosConfig';

const onboardingService = {
  initiate: (empId, data) => api.post(`/onboarding/initiate/${empId}`, data),
  getAll: () => api.get('/onboarding'),
  getById: (id) => api.get(`/onboarding/${id}`),
  getByEmployee: (empId) => api.get(`/onboarding/employee/${empId}`),
  updateStatus: (id, status) =>
    api.patch(`/onboarding/${id}/status?status=${status}`),

  // Tasks
  addTask: (onboardingId, data) =>
    api.post(`/onboarding/${onboardingId}/tasks`, data),
  updateTask: (taskId, data) =>
    api.put(`/onboarding/tasks/${taskId}`, data),
  completeTask: (taskId) =>
    api.patch(`/onboarding/tasks/${taskId}/complete`),
  deleteTask: (taskId) =>
    api.delete(`/onboarding/tasks/${taskId}`),
  getTasks: (onboardingId) =>
    api.get(`/onboarding/${onboardingId}/tasks`),

  getStatistics: () => api.get('/onboarding/statistics'),
};

export default onboardingService;

File: hrms-backend/src/main/java/com/hrms/service/OnboardingService.java

//service/OnboardingService.java
package com.hrms.service;

import com.hrms.model.OnboardingProcess;
import com.hrms.model.OnboardingTask;
import java.util.List;
import java.util.Map;

public interface OnboardingService {
 OnboardingProcess initiate(Long employeeId, OnboardingProcess process);
 OnboardingProcess getById(Long id);
 OnboardingProcess getByEmployee(Long employeeId);
 List<OnboardingProcess> getAll();
 OnboardingProcess updateStatus(Long id, String status);

 // Tasks
 OnboardingTask addTask(Long onboardingId, OnboardingTask task);
 OnboardingTask updateTask(Long taskId, OnboardingTask task);
 OnboardingTask completeTask(Long taskId);
 void deleteTask(Long taskId);
 List<OnboardingTask> getTasks(Long onboardingId);

 // Templates - Create default tasks
 void createDefaultTasks(OnboardingProcess process);

 Map<String, Object> getStatistics();
}